examples/automatic_speech_recognition/automatic_speech_recognition/evaluator.py

```diff
@@ -0,0 +1,301 @@
+# Copyright 2021-2023 Kolena Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List
+from typing import Optional
+from typing import Tuple
+from typing import Union
+
+import langcodes
+import langid
+import numpy as np
+from jiwer import cer
+from jiwer import process_words
+from numwords_to_nums.numwords_to_nums import NumWordsToNum
+from workflow import GroundTruth
+from workflow import Inference
+from workflow import TestCase
+from workflow import TestCaseMetric
+from workflow import TestSample
+from workflow import TestSampleMetric
+from workflow import TestSuiteMetric
+
+from kolena.workflow import AxisConfig
+from kolena.workflow import Curve
+from kolena.workflow import CurvePlot
+from kolena.workflow import Histogram
+from kolena.workflow import Plot
+from kolena.workflow.evaluator_function import EvaluationResults
+from kolena.workflow.evaluator_function import TestCases
+from kolena.workflow.annotation import ClassificationLabel
+
+from utils import generate_diff_word_level, preprocess_transcription
+
+
+def compute_test_sample_metrics(gt: GroundTruth, inf: Inference) -> TestSampleMetric:
+    gt = preprocess_transcription(gt)
+    inf = preprocess_transcription(inf)
+
+    word_fp, (fn_count, fp_count, ins_count, del_count, sub_count, sub_list, ins_list, del_list) = generate_diff_word_level(gt, inf, mode="fp")
```

Comment:
Instead of having `mode`, can we have the util function output both `word_fp` and `word_fn`?